### DISCUSSION TITLE
Setting up training for multiple concepts

### DISCUSSION THREAD

**Discussion by Jonseed** [d8ahazard/sd_dreambooth_extension#131] [Q&A]
A few questions about training multiple concepts concurrently (if there are answers):

- How do you format the JSON file if you are training multiple concepts concurrently? Just a comma between the next set of curly brackets?
- How different should a concept be to be trained as a separate concept? Should it be like a completely different character, or could it be the same character in a different set of clothes?
- I'm guessing the instance prompts would also come from the filename.txt files in each dataset if that option is selected?
- Can the class dataset be the same (since it is more generic anyway and may apply to all concepts)? Should you have a 10X multiple of all the training images combined? So if you have 2 concepts, each with 20 photos, would you have 400 class images? Or is 200 sufficient?
- Similar to the above question, should the training steps me a 100X multiple of all your training images combined? So if you have 2 concepts, each with 20 photos, your training steps should be about 4000 steps?
- Should each concept have a unique instance keyword in the txt files, in addition to the descriptive captions?
- Should it be considered that the concepts are basically trained completely separate from each other, but just included in the same model file, and must each be invoked independently in the prompt when inferencing?

I'm sure I'll have more questions...

**Comment by Jonseed** [comment]
Any thoughts?

**Comment by mekrod** [comment]
i am not an expert yet, i only trained a few models, but here is what i learned:
- How do you format the JSON file if you are training multiple concepts concurrently? Just a comma between the next set of curly brackets?
yes

- How different should a concept be to be trained as a separate concept? Should it be like a completely different character, or could it be the same character in a different set of clothes?
 it depends of your goal, if you want to capture a drawing style for example, you can train with any drawing as longs as it follows the style, however if you want to capture some specific elements of this style you need to separate them on different concepts, if you want to train a subject like a person of an object, it is better if the subject is in different angles, lighting and clothing.

- I'm guessing the instance prompts would also come from the filename.txt files in each dataset if that option is selected?
they will come from the file instead of the instance prompt field

- Can the class dataset be the same (since it is more generic anyway and may apply to all concepts)? Should you have a 10X multiple of all the training images combined? So if you have 2 concepts, each with 20 photos, would you have 400 class images? Or is 200 sufficient?
 i'm not sure here but i believe that you can only regulate 1 class (so you need to regulate the class that you stablished on the class field, not the concepts), but i can be wrong here

- Similar to the above question, should the training steps me a 100X multiple of all your training images combined? So if you have 2 concepts, each with 20 photos, your training steps should be about 4000 steps?
 yeah, but remember that is an aproximation, it works well most of the times anyways. 

- Should each concept have a unique instance keyword in the txt files, in addition to the descriptive captions?
yes

- Should it be considered that the concepts are basically trained completely separate from each other, but just included in the same model file, and must each be invoked independently in the prompt when inferencing?
idk if they are trained separately but every caption counts when prompting, even the ones you use to describe the subject

**Comment by leoftm** [comment]
I'm not sure I'm using filename.txt in the right way.
If filename.txt is generated using deepdanbooru in the image, what is contained in it is needed for class prompt and
Is not the missing unlearned tag+filename.txt needed for Instance prompt?

https://dreambooth.github.io
https://dreambooth.github.io/DreamBooth_files/system.png
When I saw this I thought.

ex)model doesn't know the word "yup".
prompt generated by deepdanbooru with yup picture
1girl,black hair,hogehoge
instance prompt: yup,1girl,black hair,hogehoge
class prompt:1girl,black hair,hogehoge

Now Dreambooth's filename.txt is
"Uses the filename.txt file's content as the image labels instead of the instance prompt". "Uses the filename.txt file's contents as the image labels instead of the instance prompt".
However, there are no options in the class prompt.

Am I understanding this wrong?I don't even know if this is a topic I should be writing about here:)
